Shared/Samples/Generate offline map with custom parameters/GenerateOfflineMapWithCustomParametersView.Model.swift

```diff
@@ -0,0 +1,308 @@
+            offlineMap.initialViewpoint = Viewpoint(boundingGeometry: extent.expanded(by: 0.8))
+        }
+        
+        /// Cancels the generate offline map job.
+        func cancelJob() async {
+            // Cancels the generate offline map job.
+            await generateOfflineMapJob?.cancel()
+            generateOfflineMapJob = nil
+        }
+        
+        /// Creates a temporary directory.
+        private static func createTemporaryDirectory() -> URL {
+            // swiftlint:disable:next force_try
+            try! FileManager.default.url(
+                for: .itemReplacementDirectory,
+                in: .userDomainMask,
+                appropriateFor: FileManager.default.temporaryDirectory,
+                create: true
+            )
+        }
+        
+        // MARK: - Basemap helpers
+        private func getExportTileCacheParametersForBasemapLayer() -> ExportTileCacheParameters? {
+            if let basemapLayer = onlineMap.basemap?.baseLayers.first as? Layer {
+                if let key = OfflineMapParametersKey(layer: basemapLayer) {
+                    return offlineMapParameterOverrides?.exportTileCacheParameters[key]
+                }
+            }
+            return nil
+        }
+        
+        /// Sets the scale level range so that only the levels between the min and max inclusive,
+        /// are downloaded. Note that lower values are zoomed further out,
+        /// i.e. 0 has the least detail, but one tile covers the entire Earth.
+        /// Parameters:
+        /// - minScaleLevel: The minimum scale level to download for the basemap.
+        /// - maxScaleLevel: The maximum scale level to download for the basemap.
+        func restrictBasemapScaleLevelRangeTo(minScaleLevel: Double, maxScaleLevel: Double) {
+            guard let tileCacheParameters = getExportTileCacheParametersForBasemapLayer(),
+                  // Ensure that the lower bound of the range is not greater than the upper bound.
+                  minScaleLevel <= maxScaleLevel else {
+                return
+            }
+            
+            let scaleLevelIDs = Array(Int(minScaleLevel)...Int(maxScaleLevel))
+            // Override the default level IDs.
+            tileCacheParameters.removeAllLevelIDs()
+            tileCacheParameters.addLevelIDs(scaleLevelIDs)
+        }
+        
+        /// Adds extra padding to the extent envelope to fetch a larger area, in meters.
+        /// - Parameter basemapExtentBufferDistance: The distance to extend the area of interest.
+        func bufferBasemapAreaOfInterest(by basemapExtentBufferDistance: Double) {
+            guard let tileCacheParameters = getExportTileCacheParametersForBasemapLayer(),
+                  // The area initially specified for download when the default parameters object
+                  // was created.
+                    let areaOfInterest = tileCacheParameters.areaOfInterest else {
```

Comment:
```suggestion
                  let areaOfInterest = tileCacheParameters.areaOfInterest else {
```
I think this is an Xcode indentation bug